cmd/registry/patch/patch_project_test.go

```diff
@@ -240,3 +240,100 @@ func TestProjectExport(t *testing.T) {
 		})
 	}
 }
+
+func TestApiExport(t *testing.T) {
+	tests := []struct {
+		desc string
+		root string
+	}{
+		{
+			desc: "sample-hierarchical",
+			root: "testdata/sample-hierarchical",
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.desc, func(t *testing.T) {
+			ctx := context.Background()
+			adminClient, err := connection.NewAdminClient(ctx)
+			if err != nil {
+				t.Fatalf("Setup: failed to create client: %+v", err)
+			}
+			defer adminClient.Close()
+			project := names.Project{ProjectID: "patch-export-api-test"}
+			if err = adminClient.DeleteProject(ctx, &rpc.DeleteProjectRequest{
+				Name:  project.String(),
+				Force: true,
+			}); err != nil && status.Code(err) != codes.NotFound {
+				t.Errorf("Setup: failed to delete test project: %s", err)
+			}
+
+			if _, err := adminClient.CreateProject(ctx, &rpc.CreateProjectRequest{
+				ProjectId: project.ProjectID,
+				Project:   &rpc.Project{},
+			}); err != nil {
+				t.Fatalf("Setup: Failed to create test project: %s", err)
+			}
+
+			// set the configured registry.project to the test project
+			config, err := connection.ActiveConfig()
+			if err != nil {
+				t.Fatalf("Setup: Failed to get registry configuration: %s", err)
+			}
+			config.Project = project.ProjectID
+			connection.SetConfig(config)
+
+			registryClient, err := connection.NewRegistryClient(ctx)
+			if err != nil {
+				t.Fatalf("Setup: Failed to create registry client: %s", err)
+			}
+			defer registryClient.Close()
+
+			if err := Apply(ctx, registryClient, test.root+"/apis/registry", project.String()+"/locations/global", true, 1); err != nil {
+				t.Fatalf("Apply() returned error: %s", err)
+			}
+
+			tempDir, err := os.MkdirTemp("", "sample-export-")
+			if err != nil {
+				t.Fatalf("Setup: Failed to create export directory: %s", err)
+			}
+			defer os.RemoveAll(tempDir)
+
+			taskQueue, wait := core.WorkerPool(ctx, 1)
+			err = ExportAPI(ctx, registryClient, project.Api("registry"), tempDir, taskQueue)
+			if err != nil {
+				t.Fatalf("Setup: Failed to export api: %s", err)
+			}
+			wait()
+
+			// does the exported directory contain the API description?
+			if err := filepath.Walk(test.root+"/apis/registry", func(path string, info os.FileInfo, err error) error {
+				if info.IsDir() {
+					return nil
+				}
+				refBytes, err := os.ReadFile(path)
+				if err != nil {
+					return err
+				}
+				path = strings.TrimPrefix(path, test.root)
+				newFilename := filepath.Join(tempDir, project.ProjectID, path)
+				newBytes, err := os.ReadFile(newFilename)
+				if err != nil {
+					return err
+				}
+				if diff := cmp.Diff(newBytes, refBytes); diff != "" {
+					t.Errorf("mismatched export %s %+v", newFilename, diff)
+				}
+				return nil
+			}); err != nil {
+				t.Fatalf("Setup: Failed to export api: %s", err)
+			}
+
+			if err := adminClient.DeleteProject(ctx, &rpc.DeleteProjectRequest{
```

Comment:
This should be put into a `t.Cleanup()` or it may not run.

Reply:
thanks, fixed